src/main/java/github/scarsz/discordsrv/api/ApiManager.java

```diff
@@ -174,15 +177,26 @@ public void updateSlashCommands() {
         }
         slashCommandProviders.forEach(p -> commands.addAll(p.getSlashCommands()));
 
+        int conflictingCommands = 0;
+        Map<String, PluginSlashCommand> conflictResolvedCommands = new HashMap<>();
+        for (PluginSlashCommand pluginSlashCommand : commands) {
+            String name = pluginSlashCommand.getCommandData().getName();
+            PluginSlashCommand conflictingCommand = conflictResolvedCommands.putIfAbsent(name, pluginSlashCommand);
+            if (conflictingCommand == null) continue;
+            conflictingCommands++;
+            if (pluginSlashCommand.getPriority().ordinal() < conflictingCommand.getPriority().ordinal())
+                conflictResolvedCommands.put(name, pluginSlashCommand);
+        }
```

Comment:
Non conflicting commands are never actually added to `conflictResolvedCommands`

Reply:
That should be correct, `conflictResolvedCommands` is the set of commands from `commands` but with the conflicting ones removed according to their priority.

Reply:
Nevermind me I thought putIfAbsent was using a different collection